usr/local/www/system_firmware.php

```diff
@@ -202,116 +197,128 @@ function file_upload_error_message($error_code) {
-?>
-			</td>
-		</tr>
-		<tr>
-			<td>
-				<div id="mainarea">
-					<table class="tabcont" width="100%" border="0" cellpadding="6" cellspacing="0" summary="main area">
-					<tr>
-						<td colspan="2" class="listtopic"><?=gettext("Invoke") ." ". $g['product_name'] ." ".  gettext("Manual Upgrade"); ?></td>
-					</tr>
-					<tr>
-						<td width="22%" valign="baseline" class="vncell">&nbsp;</td>
-						<td width="78%" class="vtable">
-						<?php if (!is_subsystem_dirty('rebootreq')): ?>
-						<?php if (!is_subsystem_dirty('firmware')): ?>
-						<input name="Submit" type="submit" class="formbtn" value="<?=gettext("Enable firmware upload");?>" />
-						<br />
-							<?php printf(gettext('Click "Enable firmware upload" to begin.'),$g['firmware_update_text']);?>
-						<br />
-						<?php else: ?>
-							<input name="Submit" type="submit" class="formbtn" value="<?=gettext("Disable firmware upload");?>" />
-					</td>
-					</tr>
-					<tr>
-						<td width="22%" valign="baseline" class="vncell">&nbsp;</td>
-						<td width="78%" class="vtable">
-							<?php
-								if ($g['platform'] == "nanobsd")
-									$type = "*.img.gz";
-								else
-									$type = "*.tgz";
-							?>
-							<strong><?=gettext("Firmware image file ($type):");?> </strong>
-							<input name="ulfile" type="file" class="formfld" />
-							<br />
-							<?php if ($g['hidebackupbeforeupgrade'] === false): ?>
-							<input type="checkbox" name='backupbeforeupgrade' id='backupbeforeupgrade' /> <?=gettext("Perform full backup prior to upgrade");?>
-							<br />
-							<?php endif; ?>
-							<input name="Submit" type="submit" class="formbtn" value="<?=gettext("Upgrade firmware");?>" />
-							<?=gettext('Click "Upgrade firmware" to start the upgrade process.');?>
-						<?php endif; else: ?>
-							<strong><?=gettext("You must reboot the system before you can upgrade the firmware.");?></strong>
-						<?php endif; ?>
-				</td>
-			</tr>
-			<tr>
-				<td width="22%" valign="top">&nbsp;</td>
-				<td width="78%">
-						<?php if (is_subsystem_dirty('firmware')): ?>
-					<span class="vexpl">
-						<span class="red">
-							<strong>
-								<?=gettext("Warning:");?><br />
-							</strong>
-						</span>
-						<?=gettext("DO NOT abort the firmware upgrade once it " .
-								"has started. The firewall will reboot automatically after " .
-								"storing the new firmware. The configuration will be maintained.");?>
-					</span>
-						<?php endif; ?>
-				</td></tr>
-			</table>
-		</div>
-	</td></tr>
-</table>
-
-<?php endif; endif; ?>
-<?php include("fend.inc"); ?>
-</form>
-</body>
-</html>
+
+} else {
+	// This is where the work gets done so Forms.classes will be used from this point
+	if (!is_subsystem_dirty('firmwarelock')) {
+		require('classes/Form.class.php');
+
+		if (!is_subsystem_dirty('rebootreq')) {
+			if (!is_subsystem_dirty('firmware')) {
+				$enablebtn = new Form_Button(Submit, 'Enable firmware upload', null);
```

Comment:
Please use proper newlines here. Also, *Submit* should be quoted as its a string. And the 3rd parameter is null by default, so please omit it